fe/fe-core/src/test/java/com/starrocks/planner/MaterializedViewMultiJoinTest.java

```diff
@@ -349,4 +354,87 @@ public void testPartitionPrune2() throws Exception {
         starRocksAssert.dropMaterializedView("test_mv2");
     }
 
+    @Test
+    public void testRuleExhausted_SingleTable() throws Exception {
+        String mvName = "test_exhaused";
+        starRocksAssert.withMaterializedView("CREATE MATERIALIZED VIEW " + mvName + "\n" +
+                "REFRESH MANUAL\n" +
+                "PROPERTIES (\n" +
+                "\"replication_num\"=\"1\"\n" +
+                ")\n" +
+                "AS SELECT " +
+                "sum(p1.p1_col4) AS p1_col4\n" +
+                "FROM " +
+                "tbl_1 AS p1 ");
+        String query = "select sum(p1.p1_col4) from tbl_1 p1";
+        starRocksAssert.query(query).explainContains(mvName);
+
+        // Exhaust the rule
+        new MockUp<OptimizerContext>() {
+            @Mock
+            public boolean ruleExhausted(RuleType ruleType) {
+                return true;
+            }
+        };
+        starRocksAssert.query(query).explainWithout(mvName);
+    }
+
+    @Test
+    public void testRuleExhausted_MultiTable() throws Exception {
```

Comment:
use camel

Reply:
This kind of naming style indicates `TestPurpose_TestParameter`:

https://www.baeldung.com/java-unit-testing-best-practices